ckan/logic/action/delete.py

```diff
@@ -346,12 +346,17 @@ def _group_or_org_delete(context, data_dict, is_org=False):
     else:
         _check_access('group_delete', context, data_dict)
 
-    # organization delete will delete all datasets for that org
-    # FIXME this gets all the packages the user can see which generally will
-    # be all but this is only a fluke so we should fix this properly
+    # organization delete will not occure whilke all datasets for that org are
+    # not deleted
     if is_org:
-        for pkg in group.packages(with_private=True):
-            _get_action('package_delete')(context, {'id': pkg.id})
+        datasets = model.Session.query(model.Package) \
+                        .filter_by(owner_org=group.id) \
+                        .filter(model.Package.state != 'deleted') \
+                        .count()
+        if datasets:
+            if not authz.check_config_permission('ckan.auth.create_unowned_dataset'):
```

Comment:
If `ckan.auth.create_unowned_dataset=True` I think we should update the `owner_org` value of these datasets to None. Ideally users should be warned of this when clicking Delete, @wardi @smotornyuk @tino097 WDYT?

Reply:
thats right, ive updated it to None